project.clj

```diff
@@ -87,10 +87,11 @@
 
   :clean-targets ["target"]
 
-  :aliases {"shadow-build" ["shell" "sh" "-c" "npm install && npx shadow-cljs compile app"]
-            "shadow-release" ["shell" "sh" "-c" "npm install && npx shadow-cljs release app"]
-            "shadow-test" ["shell" "sh" "-c" "npm install --include=dev && npx shadow-cljs compile cljs-test && ./node_modules/karma/bin/karma start"]
-            "shadow-watch" ["shell" "sh" "-c" "npm install --include=dev && npx shadow-cljs watch app"]
+  :aliases {"npm-deps" ["shell" "sh" "-c" "npm install --save=false --package-lock=true"] ; npm ci always removes node_modules by design, which is unnecessary and time consuming
+            "shadow-build" ["do" ["npm-deps"] ["shell" "sh" "-c" "npx shadow-cljs compile app --config-merge '{:build-options {:cache-level :jars}}'"]]
```

Comment:
`{:build-options {:cache-level :jars}}` described more here: https://shadow-cljs.github.io/docs/UsersGuide.html#_compiler_cache

here so we can save only dependencies from shadow-cljs build cache